src/main/java/reposense/system/DashboardServer.java

```diff
@@ -45,7 +45,7 @@ public static void startServer(int port, Path requestPath) {
      */
     private static void launchBrowser(String url) throws IOException {
         try {
-            if (Desktop.isDesktopSupported()) {
+            if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
```

Comment:
just to check, with the added condition, is the previous one `Desktop.isDesktopSupported()` still needed?

Reply:
It is needed, otherwise, exception will be thrown during `Desktop.getDesktop()` if the desktop is not supported.